src/test/java/org/opensearch/jobscheduler/transport/GetJobTypeTransportActionTests.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.jobscheduler.transport;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.opensearch.action.ActionListener;
+import org.opensearch.action.support.ActionFilters;
+import org.opensearch.tasks.Task;
+import org.opensearch.test.OpenSearchTestCase;
+import org.opensearch.transport.TransportService;
+
+import static org.mockito.Mockito.mock;
+
+public class GetJobTypeTransportActionTests extends OpenSearchTestCase {
+
+    private GetJobTypeTransportAction action;
+
+    private Task task;
+
+    private GetJobTypeRequest request;
+
+    private ActionListener<GetJobDetailsResponse> response;
+
+    @Before
+    public void setUp() throws Exception {
+        super.setUp();
+        action = new GetJobTypeTransportAction("", mock(TransportService.class), mock(ActionFilters.class));
+        request = new GetJobTypeRequest("demo_job_type", "extension_id");
+        task = mock(Task.class);
+        response = new ActionListener<GetJobDetailsResponse>() {
+            @Override
+            public void onResponse(GetJobDetailsResponse jobDetailsResponse) {
+                // onResponse will not be called as we do not have the AD index
+                Assert.assertEquals("success", jobDetailsResponse.getResponse());
```

Comment:
If this isn't supposed to be called why not do `fail("We do not have index so onResponse should not be called");`

Reply:
This assertion is executed from the testPrepareRequest. The execution flow is when the call is made and then response is returned and handled by action listener

Reply:
I'm not sure I understand the comment on line 40, then.    Can we expect `onResponse()` to be called?  Can we expect `onFailure()` to not be called (in which case we should put a `fail()` call there.)

Reply:
ACk